crates/asm/subprotocols/checkpoint/src/state.rs

```diff
@@ -46,21 +67,31 @@ impl CheckpointState {
     }
 
     /// Returns the verified epoch summary, if any.
+    ///
+    /// Returns `None` if no checkpoint has been verified yet (genesis state).
     pub fn verified_epoch_summary(&self) -> Option<&EpochSummary> {
-        self.verified_epoch_summary.as_ref()
+        if self.is_genesis() {
+            None
+        } else {
+            Some(&self.epoch_summary)
+        }
     }
```

Comment:
Can you provide more info on this function?
I'd assume that we store "locally" the (non-verified) EpochSummary and separately, after reading it from L1 store an indicator that the EpochSummary is verified (if valid), but I dee that now we no longer have it in `CheckpointState`.

Do I understand correctly that because it's part of the checkpoint subprotocol, we only get access to the checkpoint, here, after it's on L1?